File: sdks/guides/python/routing/mock_classifier.py
"""Mock classifier - no external API dependencies."""


def mock_classify(message: str) -> str:
    lower = message.lower()
    if any(w in lower for w in ("bug", "error", "help")):
        return "support"
    if any(w in lower for w in ("price", "buy", "plan")):
        return "sales"
    return "other"


def mock_reply(message: str, role: str) -> str:
    if role == "support":
        return f"[Support] I can help with that technical issue. Let me look into: {message}"
    if role == "sales":
        return f"[Sales] Great question about pricing! Here's what I can tell you about: {message}"
    return f"[General] Thanks for reaching out. Regarding: {message}"

File: sdks/guides/go/routing/mock_classifier.go
package main

import "strings"

func MockClassify(message string) string {
	lower := strings.ToLower(message)
	for _, w := range []string{"bug", "error", "help"} {
		if strings.Contains(lower, w) {
			return "support"
		}
	}
	for _, w := range []string{"price", "buy", "plan"} {
		if strings.Contains(lower, w) {
			return "sales"
		}
	}
	return "other"
}

func MockReply(message, role string) string {
	switch role {
	case "support":
		return "[Support] I can help with that technical issue. Let me look into: " + message
	case "sales":
		return "[Sales] Great question about pricing! Here's what I can tell you about: " + message
	default:
		return "[General] Thanks for reaching out. Regarding: " + message
	}
}
